### PR TITLE
17334 autocompletion in description and comments

### PR DESCRIPTION
**Pull request by hschink**
[`* `#17334 Work package auto-completion for description and comments`](https://community.openproject.org/work_packages/17334)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://fix-17334-autocompletion-in-description-and-comments.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

**Comment by myabc** [comment]
@hschink ~~This isn't working for me at all (no errors, just fails silently). Tried in both FF and Chrome. I'm not sure if something is wrong with my local env.~~

File: app/assets/javascripts/angular/helpers/auto-complete-helper.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          return items; // we do not sort
+        }
+      }
+    };
+  };
+
+  return {
+    enableTextareaAutoCompletion: function(textareas) {
+      angular.forEach(textareas, function(textarea) {
+        var url = jQuery(textarea).data('wp_autocomplete_url');
```
Comment (myabc) [inline app/assets/javascripts/angular/helpers/auto-complete-helper.js]: @hschink Let's standardise on `angular.element`.

**Comment by myabc** [comment]
@hschink the first HTTP request fires after entering an `#`. Would it be better to wait until a second character (i.e. a digit) is entered? Do you know of a way to customise this behaviour?

**Comment by myabc** [comment]
:+1:

**Pull request merged**